hudi-common/src/main/java/org/apache/hudi/common/table/timeline/versioning/v1/ArchivedTimelineV1.java

```diff
@@ -274,6 +308,39 @@ public Map<String, List<HoodieInstant>> getInstantsInRangeCollected() {
     }
   }
 
+  public class InstantsLoaderWithLimit implements BiConsumer<String, GenericRecord> {
+    private final Map<String, List<HoodieInstant>> instantsInRange = new ConcurrentHashMap<>();
+    private final boolean loadInstantDetails;
+    private final int limit;
+    private volatile int loadedCount = 0;
+
+    private InstantsLoaderWithLimit(boolean loadInstantDetails, int limit) {
+      this.loadInstantDetails = loadInstantDetails;
+      this.limit = limit;
+    }
+
+    @Override
+    public void accept(String instantTime, GenericRecord record) {
+      if (loadedCount >= limit) {
+        return;
```

Comment:
even though it returns here, the left over files/entries are still read and decoded.